laravel-live-coding/app/Http/Controllers/ReviewController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\ReviewResource;
+use App\Services\ReviewService;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class ReviewController extends Controller
+{
+    private ReviewService $reviewService;
+
+    public function __construct(ReviewService $reviewService)
+    {
+        $this->reviewService = $reviewService;
+    }
```

Comment:
have to use abstraction (interface) "D" Principle and if you have to change realization you have to change this class and you broke "O" principle